src/main/java/net/greatstart/controllers/InvestmentController.java

```diff
@@ -1,31 +1,108 @@
+        this.projectService = projectService;
+        this.userService = userService;
     }
 
-    public Investment saveInvestment(Investment investment) {
-        return investmentService.saveInvestment(investment);
+    @GetMapping("/project/{id}/addInvestment")
+    public ModelAndView getAddInvestmentForm(@PathVariable long id) {
+        Project project = projectService.getProjectById(id);
+
+        ModelAndView modelAndView = new ModelAndView("investment/add_investment");
+        modelAndView.addObject("investment", new Investment());
+        modelAndView.addObject(project);
+        return modelAndView;
+    }
+
+    @PostMapping("/project/{id}/addInvestment")
+    public ModelAndView addInvestment(@PathVariable long id,
+                                      @RequestParam BigDecimal sum,
+                                      Principal principal) {
+        User investor = userService.getUserByEmail(principal.getName());
+        if (investor.getContact().getPhoneNumber() == null
+                || investor.getContact().getPhoneNumber().isEmpty()) {
+            ModelAndView modelAndView = new ModelAndView(REDIRECT + EDIT_USER_PROFILE);
+            modelAndView.addObject("message", "Please enter your phone number!");
+            return modelAndView;
+        }
+
+        Project project = projectService.getProjectById(id);
+        String message = InvestmentValidationService.validate(sum, project);
+        if (message != null) {
+            ModelAndView modelAndView = new ModelAndView("/project/" + id + "/addInvestment");
+            modelAndView.addObject("message", message);
+            return modelAndView;
+        }
+        investmentService.saveInvestment(new Investment(LocalDateTime.now(),
+                project, investor, sum, false, false));
+
+        return new ModelAndView(REDIRECT + PROJECT_PAGE + id);
+    }
+
+    @GetMapping("investment/{id}")
+    public Investment getInvestmentById(@PathVariable long id) {
+        return investmentService.getInvestmentById(id);
+    }
+
+    @GetMapping("investment")
+    public ModelAndView getAllInvestments() {
+        ModelAndView modelAndView = new ModelAndView(INVESTMENTS_VIEW);
+        modelAndView.addObject("investmentList",
+                investmentService.getAllInvestments());
+        return modelAndView;
     }
 
-    public Investment getInvestmentById(long id) {
-        return investmentService.getnvestmentById(id);
+    @GetMapping("project/{id}/investments")
+    public List<Investment> getAllProjectInvestments(@PathVariable long id) {
+        return projectService.getProjectById(id).getInvestments();
     }
 
-    public List<Investment> getAllnvestments() {
-        return investmentService.getAllInvestments();
+    @GetMapping("user/{id}/investments")
+    public List<Investment> getAllUserInvestments(Principal principal) {
+        return userService.getUserByEmail(principal.getName()).getInvestments();
+    }
+
+    //todo
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1135)